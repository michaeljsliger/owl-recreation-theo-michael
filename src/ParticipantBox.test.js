import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ParticipantBox from './ParticipantBox';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Participant Box Branch Testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantBox />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<ParticipantBox />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})