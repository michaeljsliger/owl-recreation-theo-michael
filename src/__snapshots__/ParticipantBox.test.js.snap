// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Box Branch Testing should render correctly 1`] = `
<ul>
  <li>
    <div
      className="sessionParticipant"
    >
      <img
        alt="avatar"
        className="user-avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <div
        className="participant-info"
      >
        <h6
          className="particpant-name"
        >
          Koren Templeton
        </h6>
      </div>
      <div
        className="participant-status"
      />
    </div>
  </li>
  <li>
    <div
      className="sessionParticipant"
    >
      <img
        alt="avatar"
        className="user-avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <div
        className="participant-info"
      >
        <h6
          className="particpant-name"
        >
          Caty Flucker
        </h6>
      </div>
      <div
        className="participant-status"
      />
    </div>
  </li>
  <li>
    <div
      className="sessionParticipant"
    >
      <img
        alt="avatar"
        className="user-avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <div
        className="participant-info"
      >
        <h6
          className="particpant-name"
        >
          Frank Runciman
        </h6>
      </div>
      <div
        className="participant-status"
      />
    </div>
  </li>
  <li>
    <div
      className="sessionParticipant"
    >
      <img
        alt="avatar"
        className="user-avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <div
        className="participant-info"
      >
        <h6
          className="particpant-name"
        >
          Ashla Attwool
        </h6>
      </div>
      <div
        className="participant-status"
      />
    </div>
  </li>
</ul>
`;
